packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -435,6 +443,12 @@ export class Repository extends RepositoryBase {
       resourceName: this.physicalName,
     });
 
+    // repository image tag mutability
+    if (props.imageTagMutability !== undefined) {
+      this.imageTagMutability = props.imageTagMutability;
+      resource.imageTagMutability = props.imageTagMutability;
+    }
+
```

Comment:
I'd say we should just move passing this property to the L1 resource up into the constructor. Like
```typescript
       const resource = new CfnRepository(this, 'Resource', {
      repositoryName: this.physicalName,
      // It says "Text", but they actually mean "Object".
      repositoryPolicyText: Lazy.anyValue({ produce: () => this.policyDocument }),
      lifecyclePolicy: Lazy.anyValue({ produce: () => this.renderLifecyclePolicy() }),
      imageTagMutability: props.imageTagMutability,
    });
```

If its undefined the property won't be included in synth output.

Reply:
agreed, will make this changes soon, thanks.